packit_service/worker/handlers/distgit.py

```diff
@@ -1015,3 +1016,83 @@ def get_trigger_type_description(self) -> str:
             f"Fedora Koji build was re-triggered "
             f"by comment in issue {self.data.issue_id}."
         )
+
+
+@configured_as(job_type=JobType.koji_build)
+@run_for_comment(command="koji-tag")
+@reacts_to(event=PullRequestCommentPagureEvent)
+class TagIntoSidetagHandler(
+    RetriableJobHandler,
+    ConfigFromEventMixin,
+    PackitAPIWithDownstreamMixin,
+    GetPagurePullRequestMixin,
+):
+    task_name = TaskName.tag_into_sidetag
+
+    _koji_helper: Optional[KojiHelper] = None
+
+    @property
+    def koji_helper(self):
+        if not self._koji_helper:
+            self._koji_helper = KojiHelper()
+        return self._koji_helper
+
+    @staticmethod
+    def get_checkers() -> Tuple[Type[Checker], ...]:
+        return (PermissionOnDistgit,)
+
+    def run_for_branch(self, job_config: JobConfig, branch: str) -> None:
+        sidetag_group = SidetagGroupModel.get_or_create(job_config.sidetag_group)
+        sidetag = SidetagModel.get_or_create(sidetag_group, branch)
+        # we need Kerberos ticket to tag a build into sidetag
+        # and to create a new sidetag (if needed)
+        self.packit_api.init_kerberos_ticket()
+        if not sidetag.koji_name or not self.koji_helper.get_tag_info(
+            sidetag.koji_name
+        ):
+            tag_info = self.koji_helper.create_sidetag(branch)
+            if not tag_info:
+                logger.error(f"Failed to create sidetag for {branch}")
+                return
+            sidetag.set_koji_name(tag_info["name"])
+        if not (
+            nvr := self.koji_helper.get_latest_stable_nvr(
+                job_config.downstream_package_name, branch
+            )
+        ):
+            logger.debug(
+                "Failed to get the latest stable build "
+                f"of {job_config.downstream_package_name} for {branch}"
+            )
+            return
+        logger.debug(f"Tagging {nvr} into {sidetag.koji_name}")
+        self.koji_helper.tag_build(nvr, sidetag.koji_name)
+
+    def run(self) -> TaskResults:
+        comment = self.data.event_dict.get("comment")
+        commands = get_packit_commands_from_comment(
+            comment, self.service_config.comment_command_prefix
+        )
+        args = commands[1:] if len(commands) > 1 else ""
+        for job in self.package_config.get_job_views():
+            if (
+                job.type == JobType.koji_build
+                and job.sidetag_group
+                and job.dist_git_branches
+            ):
+                configured_branches = aliases.get_branches(
+                    *job.dist_git_branches, default_dg_branch="rawhide"
+                )
+                if "--all-branches" in args:
+                    branches = configured_branches
```

Comment:
I may be wrong, but if there are multiple branches in `dist_git_branches` I expect we tag them multiple times. I don't know if this could be a problem? Probably not...

Reply:
Sorry, I don't understand. Do you mean I should get rid of the `--all-branches` argument?

Reply:
No, I think it is handy. But I am wondering if there could be problems if we run it multiple times for the same branch, probably not? You probably can tell it better than me.

Reply:
Oh, I think I know what Maja means… Multiple job configs for the same branch, not sure if there is a use case for such thing :thinking:

Reply:
I guess in theory such a config could exist, but like I said it shouldn't do any harm. You can have multiple `koji_build` jobs with the same sidetag group and the same dist-git branches in different packages (monorepo config), but then the packages will most likely have different `downstream_package_name`, so Packit will be tagging a different build.